doc/source/advanced.rst

```diff
@@ -231,16 +228,31 @@ Advanced indexing with hierarchical index
 -----------------------------------------
 
 Syntactically integrating ``MultiIndex`` in advanced indexing with ``.loc`` is a
-bit challenging, but we've made every effort to do so. For example the
-following works as you would expect:
+bit challenging, but we've made every effort to do so. In general, MultiIndex
+keys take the form of tuples. For example, the following works as you would expect:
 
 .. ipython:: python
 
    df = df.T
    df
-   df.loc['bar']
    df.loc['bar', 'two']
```

Comment:
this is also a bit a dubious example IMO (try `df.loc['bar', 'A']` ..). Should we maybe recommend to do `df.loc[('bar', 'two')]` as good practice? (I know in practice it does not matter, as `df.loc[('bar', 'A')]` works just as well, but it "looks" clearer. Unless we recommend `df.loc[('bar', 'two'),]` (extra comma) which is not ambiguous I think)

Reply:
YES. showing lists just perpetuates the confusion.

Reply:
Good point, this is really not obvious. It gets confusing once we mix row and column indices inside `loc`. Also, we're getting a bit into Python syntax issues here. Parentheses are not required to specify a tuple, so `df.loc[('bar', 'two')]` is exactly identical to `df.loc['bar', 'two']`. I think we should try to relate MultiIndex usage to the standard use case of `loc` where you specify rows, columns. In fact, we should probably include the worst case scenario where both rows and columns have a MultIindex...

Not sure what to do, let's discuss.

Reply:
correct they are not required (and that's the problem). However it is much better to be explicit when using mutli-indexes.

Reply:
OK, then let's *recommend* the use of parens around tuples then. But I'm not sure recommending a nested tuple adds much to resolve the confusion (i.e. `df.loc[('bar', 'two'),]` is more confusing than `df.loc[('bar', 'two')]`, because I thought that we now distinguish between tuples and lists).

Reply:
I wouldn't even personally put in terms of recommentations... I would say that the "default" is ``df.loc[('bar', 'two'),]``, and that ``df.loc['bar', 'two']`` is a shortcut which however can lead to ambiguity.

Reply:
... although then we need to mention that ambiguity is resolved in favour of multiple levels, rather than multiple axes (yeah, there are exceptions currently, but [they](https://github.com/pandas-dev/pandas/issues/19110) are [bugs](https://github.com/pandas-dev/pandas/issues/16396))

Reply:
Yep, +1 on using `df.loc[('bar', 'two'),]` in the example itself, and mentioning `df.loc['bar', 'two']` gives the same in this case but can lead to ambiguity.

Reply:
Done